scripts/pre_commit_linter.py

```diff
@@ -473,6 +468,56 @@ def _get_expression_from_node_if_one_exists(
     return expression
 
 
+def _walk_with_gitignore(root, exclude_dirs):
+    """A walk function similar to os.walk but this would ignore the files and
+    directories which is not tracked by git. Also, this will ignore the
+    directories mentioned in exclude_dirs.
+
+    Args:
+        root: str. The path from where the function should start walking.
+        exclude_dirs: list(str). A list of dir path which should be ignored.
+
+    Yields:
+        list(str). A list of unignored files.
+    """
+    dirs, file_paths = [], []
+    for name in os.listdir(root):
+        if os.path.isdir(os.path.join(root, name)):
+            dirs.append(os.path.join(root, name))
+        else:
+            file_paths.append(os.path.join(root, name))
+
+    yield [file_path for file_path in file_paths if not _is_path_ignored(
+        file_path)]
+
+    for dir_path in dirs:
+        # Adding "/" in the end of the dir path according to the git dir path
+        # structure.
+        if (not _is_path_ignored(dir_path + '/')) and (
+                dir_path not in exclude_dirs):
+            for x in _walk_with_gitignore(dir_path, exclude_dirs):
+                yield x
+
+
+def _is_path_ignored(path_to_check):
+    """Checks whether the given path is ignored by git.
+
+    Args:
+        path_to_check: str. A path to a file or a dir.
+
+    Returns:
+        bool. Whether the given path is ignored by git.
+    """
+    command = ['git', 'check-ignore', '-q', path_to_check]
+
+    # The "git check-ignore <path>" command use to returns 0 when the path is
```

Comment:
Can you please re-phrase this comment?

Reply:
I'll surely do it. Also, it would be great if you can provide some suggestion on what I should add or a pattern in which I can re-phrase this comment! :)

Reply:
Maybe something on these grounds:
`The "git check-ignore <path>" command returns 0 when the path is ignored otherwise it returns 1. subprocess.call then returns this returncode.`

Reply:
Done!